pkg/k8sreporter/reporter.go

```diff
@@ -0,0 +1,185 @@
+	r.logNodes()
+	r.logPods("openshift-sriov-network-operator")
+	r.logPods(namespaces.Test)
+	r.logLogs(func(p *corev1.Pod) bool {
+		if !strings.HasPrefix(p.Name, "sriov-") {
+			return true
+		}
+		return false
+	})
+	r.logSriovNodeState()
+	r.logNetworkPolicies()
+}
+
+// Cleanup cleans up the current content of the artifactsDir
+func (r *KubernetesReporter) Cleanup() {
+}
+
+func (r *KubernetesReporter) logPods(namespace string) {
+	pods, err := r.clients.Pods(v1.NamespaceAll).List(metav1.ListOptions{})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to fetch pods: %v\n", err)
+		return
+	}
+
+	j, err := json.MarshalIndent(pods, "", "    ")
+	if err != nil {
+		fmt.Println("Failed to marshal pods", err)
+		return
+	}
+	fmt.Fprintln(r.dumpOutput, string(j))
+}
+
+func (r *KubernetesReporter) logNodes() {
+	nodes, err := r.clients.Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to fetch nodes: %v\n", err)
+		return
+	}
+
+	j, err := json.MarshalIndent(nodes, "", "    ")
+	if err != nil {
+		fmt.Println("Failed to marshal nodes")
+		return
+	}
+	fmt.Fprintln(r.dumpOutput, string(j))
+}
+
+func (r *KubernetesReporter) logLogs(filterPods func(*corev1.Pod) bool) {
+	pods, err := r.clients.Pods(v1.NamespaceAll).List(metav1.ListOptions{})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to fetch pods: %v\n", err)
+		return
+	}
+
+	for _, pod := range pods.Items {
+		if filterPods(&pod) {
+			continue
+		}
+		for _, container := range pod.Spec.Containers {
+
+			logs, err := r.clients.Pods(pod.Namespace).GetLogs(pod.Name, &v1.PodLogOptions{Container: container.Name}).DoRaw()
+			if err == nil {
+				fmt.Fprintf(r.dumpOutput, "Dumping logs for pod %s-%s-%s", pod.Namespace, pod.Name, container.Name)
+				fmt.Fprintln(r.dumpOutput, string(logs))
+			}
+		}
+	}
+}
+
+func (r *KubernetesReporter) logNetworkPolicies() {
+	policies := sriovv1.SriovNetworkNodePolicyList{}
+	err := r.clients.List(context.Background(),
+		&policies,
+		runtimeclient.InNamespace("openshift-sriov-network-operator"))
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to fetch network policies: %v\n", err)
+		return
+	}
+
+	j, err := json.MarshalIndent(policies, "", "    ")
+	if err != nil {
+		fmt.Println("Failed to marshal policies")
+		return
+	}
+	fmt.Fprintln(r.dumpOutput, string(j))
+}
+func (r *KubernetesReporter) logNetworks() {
+	networks := sriovv1.SriovNetworkList{}
+	err := r.clients.List(context.Background(),
+		&networks,
+		runtimeclient.InNamespace("openshift-sriov-network-operator"))
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to fetch network policies: %v\n", err)
+		return
+	}
+
+	j, err := json.MarshalIndent(networks, "", "    ")
+	if err != nil {
+		fmt.Println("Failed to marshal networks")
+		return
+	}
+	fmt.Fprintln(r.dumpOutput, string(j))
+}
+
+func (r *KubernetesReporter) logSriovNodeState() {
+	nodeStates, err := r.clients.SriovNetworkNodeStates(v1.NamespaceAll).List(metav1.ListOptions{})
```

Comment:
Can we use the operator namespace here also?

Reply:
That should not make a difference, but ok